client-tez/src/main/java/org/apache/tez/runtime/library/common/shuffle/impl/RssSimpleFetchedInputAllocator.java

```diff
@@ -150,9 +202,26 @@ public synchronized FetchedInput allocate(
       long actualSize, long compressedSize, InputAttemptIdentifier inputAttemptIdentifier)
       throws IOException {
     if (actualSize > maxSingleShuffleLimit || this.usedMemory + actualSize > this.memoryLimit) {
-      LOG.info("Allocate DiskFetchedInput, length:{}", actualSize);
-      return new DiskFetchedInput(
-          actualSize + 8, inputAttemptIdentifier, this, conf, localDirAllocator, fileNameAllocator);
+      if (remoteSpillEnable) {
+        LOG.info("Allocate RemoteFetchedInput, length:{}", actualSize);
+        return new RemoteFetchedInput(
+            actualSize + 8,
+            inputAttemptIdentifier,
+            this,
+            remoteFS,
+            remoteSpillBasePath,
+            uniqueIdentifier,
+            appAttemptId);
+      } else {
+        LOG.info("Allocate DiskFetchedInput, length:{}", actualSize);
+        return new DiskFetchedInput(
+            actualSize + 8,
```

Comment:
Could use a constant variable instead of 8?

Reply:
OK, I will do it.